### PR TITLE
Added black, 

### PR DESCRIPTION
**Pull request by MeGaGiGaGon**
Also did the flake8 and isort compatibility

### CODE REVIEW

**Comment by MeGaGiGaGon** [comment]
@A5rocks For the flake8 config, should we have it as it's own file or use https://pypi.org/project/Flake8-pyproject/
Should we also move the isort config to pyproject.toml?

**Comment by A5rocks** [comment]
I guess using flake8-pyproject is nicer so we have everything in one place.

And yeah also isort.

**Comment by MeGaGiGaGon** [comment]
@A5rocks I think that should be the things, can you make sure?

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -29,3 +29,9 @@ repos:
       - id: flake8
         additional_dependencies:
           - flake8-docstrings~=1.6.0
+          - Flake8-pyproject
```
Comment (A5rocks) [inline .pre-commit-config.yaml]: might be a good idea to pin this (and tbf i prefer the casing `flake8-pyproject` but eh)

**Review by A5rocks — APPROVED** [review]
Small nitpick but you can ignore it if you want (let's discuss how to merge this in the channel when you're ready!)

**Comment by A5rocks** [comment]
(sorry about the spam, trying to get the webhook to work :( )

**Comment by MeGaGiGaGon** [comment]
Proposed merge message:

- Added support for the Black formatter on pre-commit
- Configured isort and flake8 to work with black
- Added flake8-pyproject so the flake8 configs could be moved to pyproject.toml

Still todo:

- Fix/Update module versions so that flake8-pyproject can be pinned
- Fix config settings in tox.ini

**Pull request merged**